remotespark/livyclientlib/sparkcontroller.py

```diff
@@ -73,12 +74,17 @@ def add_session(self, name, connection_string, skip_if_exists, properties):
             return
 
         session = self._create_livy_session(connection_string, properties, self.ipython_display)
+
+        spark_events = SparkEvents()
```

Comment:
Yes, I agree with you @aggFTW  to be before the call to Livy but for session creation end I think this is the right place for it .